### PR TITLE
feat(general): Use --no-fail-on-crash to gracefully exit commit_repository and setup_bridgecrew_credentials

### PR DESCRIPTION
**Pull request by rotemavni**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
In cases where `commit_repository` or `setup_bridgecrew_credentials` fail, catch the failures and exit the run with either 0 or 2 exit codes, depending on whether the user has set `--no-fail-on-crash` or not.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/main.py [hunk 1]
```diff
@@ -331,6 +335,14 @@ def run(banner: str = checkov_banner, argv: List[str] = sys.argv[1:]) -> Optiona
     git_configuration_folders = [os.getenv("CKV_GITHUB_CONF_DIR_PATH", default_github_dir_path),
                                  os.getcwd() + '/' + os.getenv('CKV_GITLAB_CONF_DIR_NAME', 'gitlab_conf')]
 
+    def commit_repository() -> Optional[str]:
+        try:
+            return bc_integration.commit_repository(config.branch)
+        except Exception as e:
+            logging.debug(f"commit_repository failed, exiting: {e}")
```
Comment (gruebel) [inline checkov/main.py:342]:
```suggestion
        except Exception:
            logging.debug(f"commit_repository failed, exiting", exc_info=True)
```

File: checkov/main.py [hunk 2]
```diff
@@ -331,6 +335,14 @@ def run(banner: str = checkov_banner, argv: List[str] = sys.argv[1:]) -> Optiona
     git_configuration_folders = [os.getenv("CKV_GITHUB_CONF_DIR_PATH", default_github_dir_path),
                                  os.getcwd() + '/' + os.getenv('CKV_GITLAB_CONF_DIR_NAME', 'gitlab_conf')]
 
+    def commit_repository() -> Optional[str]:
```
Comment (gruebel) [inline checkov/main.py:338]: you could also change it to a normal function and pass `config` as a parameter to it

**Review by nimrodkor — APPROVED** [review]
See @gruebel 's comments, nothing further
GJ!

**Review by gruebel — APPROVED** [review]
🎄

**Pull request merged**